test/dns_plugin_template_record_test.rb

```diff
@@ -1,95 +1,90 @@
+    #   MyService.expects(:create).with(:ip => '10.1.1.1', :name => 'test.example.com').returns(true)
 
-  def test_default_settings
-    Proxy::Dns::PluginTemplate::Plugin.load_test_settings({})
-    assert_equal "default_value", Proxy::Dns::Nsupdate::Plugin.settings.required_setting
-    assert_equal "/must/exist", Proxy::Dns::Nsupdate::Plugin.settings.required_path
+    assert provider.create_a_record('test.example.com', '10.1.1.1')
   end
 
-  def test_initialized_correctly
-    Proxy::Dns::PluginTemplate::Plugin.load_test_settings(:example_setting => 'a_value',
-                                                          :required_setting => 'required_setting',
-                                                          :optional_path => '/some/path',
-                                                          :required_path => '/required/path')
+  # Test A record creation fails if the record exists
+  def test_create_a_conflict
+    # Use mocha to expect any calls to backend services to prevent creating real records
+    #   MyService.expects(:create).with(:ip => '10.1.1.1', :name => 'test.example.com').returns(false)
 
-    assert_equal 'a_value', klass.new.example_setting
-    assert_equal 'required_value', klass.new.required_setting
-    assert_equal '/some/path', klass.new.optional_path
-    assert_equal '/required/path', klass.new.required_path
+    assert_raise(Proxy::Dns::Collision) { provider.create_a_record('test.example.com', '10.1.1.1') }
   end
 
-  # Test that a missing :example_setting throws an error
-  # Test A record creation
-  def test_create_a
+  # Test AAAA record creation
+  def test_create_aaaa
     # Use mocha to expect any calls to backend services to prevent creating real records
     #   MyService.expects(:create).with(:ip => '10.1.1.1', :name => 'test.example.com').returns(true)
 
-    assert klass.new.create_a_record('test.example.com', '10.1.1.1')
+    assert provider.create_aaaa_record('test.example.com', '10.1.1.1')
   end
 
   # Test A record creation fails if the record exists
-  def test_create_a_conflict
+  def test_create_aaaa_conflict
     # Use mocha to expect any calls to backend services to prevent creating real records
     #   MyService.expects(:create).with(:ip => '10.1.1.1', :name => 'test.example.com').returns(false)
 
-    assert_raise(Proxy::Dns::Collision) { klass.new.create_a_record('test.example.com', '10.1.1.1') }
+    assert_raise(Proxy::Dns::Collision) { provider.create_aaaa_record('test.example.com', '10.1.1.1') }
   end
 
   # Test PTR record creation
   def test_create_ptr
     # Use mocha to expect any calls to backend services to prevent creating real records
     #   MyService.expects(:create_reverse).with(:ip => '10.1.1.1', :name => 'test.example.com').returns(true)
 
-    assert klass.new.create_ptr_record('test.example.com', '10.1.1.1')
+    assert provider.create_ptr_record('test.example.com', '10.1.1.1')
```

Comment:
These PTR related tests are failing, it looks like the second argument should now be the PTR record (1.1.1.1.in-addr.arpa) rather than the IP address.

<pre>
Error: test_create_ptr(DnsPluginTemplateRecordTest): Proxy::Dns::Error: Not a PTR record: '10.1.1.1'
/home/dcleal/.rvm/gems/ruby-2.1.0@smart-proxy-plugin/bundler/gems/smart-proxy-350b4cf73f74/modules/dns_common/dns_common.rb:37:in `ptr_to_ip'
/home/dcleal/code/foreman/smart_proxy_dns_plugin_template/lib/smart_proxy_dns_plugin_template/dns_plugin_template_main.rb:44:in `create_ptr_record'
/home/dcleal/code/foreman/smart_proxy_dns_plugin_template/test/dns_plugin_template_record_test.rb:44:in `test_create_ptr'
</pre>